File: new-skills/references/beneficiary/edit/java/model/response/EditBeneficiaryResponse.java
/**
 * Alipay.com Inc.
 * Copyright (c) 2004-2026 All Rights Reserved.
 */
package {basePackage}.wf.model.response;

/**
 * WorldFirst editBeneficiary 响应对象。
 *
 * @author Qoder
 * @version EditBeneficiaryResponse.java, v 0.1 2026-03-26
 */
public class EditBeneficiaryResponse {

    /** API 调用结果 */
    private Result result;

    /** 收款人令牌 */
    private String beneficiaryToken;

    public Result getResult() {
        return result;
    }

    public void setResult(Result result) {
        this.result = result;
    }

    public String getBeneficiaryToken() {
        return beneficiaryToken;
    }

    public void setBeneficiaryToken(String beneficiaryToken) {
        this.beneficiaryToken = beneficiaryToken;
    }

    public boolean isSuccess() {
        return result != null && "S".equals(result.getResultStatus());
    }

    @Override
    public String toString() {
        return "EditBeneficiaryResponse{result=" + result
            + ", beneficiaryToken='" + beneficiaryToken + "'}";
    }
}

File: wf-api-integration/references/beneficiary/edit/golang/model/response/edit_beneficiary_response.go
package response

// Result represents the standard WF API result object.
type Result struct {
	ResultStatus  string `json:"resultStatus"`
	ResultCode    string `json:"resultCode"`
	ResultMessage string `json:"resultMessage"`
}

// EditBeneficiaryResponse is the response for editBeneficiary API.
type EditBeneficiaryResponse struct {
	Result           Result `json:"result"`
	BeneficiaryToken string `json:"beneficiaryToken,omitempty"`
}

